docs/source/checkpoints/dist_ckpt.rst

```diff
@@ -0,0 +1,421 @@
+
+Each saving strategy is associated with a ``backend`` and a ``version``.
+Each loading strategy can be associated with multiple ``backend``s and ``version``s it can load.
+For a given backend and version, the composition of every saving and loading strategy **must be functionally equivalent**.
+Strategies are the main way to introduce optimizations to the saving and loading algorithm without altering the checkpoint format.
+
+In the following example, the "fully parallel" wrappers modify the saving and loading *algorithm*, but the underlying checkpoint *format* (and ``backend`` in consequence) stays the same.
+It makes the ``basic_save_load`` and ``fully_parallel_save_load`` functions equivalent:
+
+.. code-block:: python
+
+    from megatron.core import dist_checkpointing
+    from megatron.core.dist_checkpointing.strategies.torch import TorchDistLoadShardedStrategy, TorchDistSaveShardedStrategy
+    from megatron.core.dist_checkpointing.strategies.fully_parallel import FullyParallelLoadStrategyWrapper, FullyParallelSaveStrategyWrapper
+
+    # Base save and load strategies defining a regular (non-parallel) save
+    base_save_strategy = TorchDistSaveShardedStrategy('torch_dist', 1)
+    base_load_strategy = TorchDistLoadShardedStrategy()
+
+    def basic_save_load(sharded_state_dict, ckpt_dir):
+        """ Save and load using some basic strategies. """
+        dist_checkpointing.save(sharded_state_dict, ckpt_dir, base_save_strategy)
+        return dist_checkpointing.load(sharded_state_dict, ckpt_dir, base_load_strategy)
+
+
+    def fully_parallel_save_load(sharded_state_dict):
+        """ Save and load using basic strategies wrapped with parallelization strategies. """
+        fully_parallel_save_strategy = FullyParallelSaveStrategyWrapper(base_save_strategy)
+        # "fully parallel" wrapper modifies the saving strategy, but not the underlying format
+        assert fully_parallel_save_strategy.backend == base_save_strategy.backend == 'torch_dist'
+        fully_parallel_load_strategy = FullyParallelLoadStrategyWrapper(base_load_strategy)
+        dist_checkpointing.save(sharded_state_dict, ckpt_dir, fully_parallel_save_strategy)
+        return dist_checkpointing.load(sharded_state_dict, ckpt_dir, fully_parallel_load_strategy)
+
+
+The ``dist_checkpointing`` package provides default strategies for some sharded backends, so it's enough to specify a tuple ``(backend, version)`` as a saving strategy.
+Backends and versions are stored in a ``metadata.json`` file inside the checkpoint so that the loading strategy can be determined automatically (provided that there exists a default loading strategy for a given backend and version).
+
+For "sharded" strategies, currently the backends supported by default are based on `PyTorch Distributed`_ format (``torch_dist`` backend) and `Zarr`_ format (``zarr`` backend).
```

Comment:
currently, if you click on "Zarr" here, it will take you to https://zarr.readthedocs.io/en/stable/.
We want to go to the Zarr section below though, right? To make the docs behave like this, we need to change the line in the Zarr section so that it uses 2 underscores after the hyperlink and backtick (ie `` `Zarr <https://zarr.readthedocs.io/en/stable/>`__  ``